src/frontend/src/binder/bind_context.rs

```diff
@@ -50,6 +50,7 @@ pub enum Clause {
     Where,
     Values,
     GroupBy,
+    On,
```

Comment:
Maybe call this `JoinOn`?

Reply:
Maybe, I will change `Filter` to `AggFilter` after I make sure it is that semantics :thinking: